test/jdk/sun/security/validator/CertReplace.java

```diff
@@ -26,22 +26,190 @@
  */
 
 import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
 import java.security.KeyStore;
+import java.security.PrivateKey;
 import java.security.cert.Certificate;
 import java.security.cert.CertificateFactory;
 import java.security.cert.X509Certificate;
-import java.util.Arrays;
 import java.util.ArrayList;
 import java.util.List;
+
+import jdk.test.lib.SecurityTools;
+import jdk.test.lib.security.KeyStoreUtils;
 import sun.security.validator.Validator;
 
+/*
+ * @test id=certreplace
+ * @bug 6948803
+ * @summary CertPath validation regression caused by SHA1 replacement root and MD2 disable feature
+ * @library /test/lib
+ * @modules java.base/sun.security.validator
+ *
+ * @run main/othervm CertReplace certreplace.jks certreplace.certs
+ */
+
+/*
+ * @test id=samedn
+ * @bug 6958869
+ * @summary Regression: PKIXValidator fails when multiple trust anchors have same dn
+ * @library /test/lib
+ * @modules java.base/sun.security.validator
+ *
+ * @run main/othervm CertReplace samedn.jks samedn1.certs
+ * @run main/othervm CertReplace samedn.jks samedn2.certs
```

Comment:
Must these tests run in `othervm`. They haven't changed any static VM properties.